pytest_fixtures/core/contenthosts.py

```diff
@@ -107,21 +107,45 @@ def registered_hosts(organization_ak_setup, content_hosts, default_sat):
 @pytest.fixture(scope="function")
 def katello_host_tools_host(default_sat, module_org, rhel_contenthost):
     """Register content host to Satellite and install katello-host-tools on the host."""
+    repo = settings.repos['SATCLIENT_REPO'][f'RHEL{rhel_contenthost.os_version.major}']
+    register_host_custom_repo(default_sat, module_org, rhel_contenthost, [repo])
+    rhel_contenthost.install_katello_host_tools()
+    yield rhel_contenthost
+
+
+@pytest.fixture(scope="function")
+def cockpit_host(default_sat, module_org, rhel_contenthost):
+    """Register content host to Satellite and install cockpit on the host."""
+    rhelver = rhel_contenthost.os_version.major
+    if rhelver > 7:
+        repo = [settings.repos[f'rhel{rhelver}_os']['baseos']]
+    else:
+        repo = [settings.repos['rhel7_os'], settings.repos['rhel7_extras']]
+    register_host_custom_repo(default_sat, module_org, rhel_contenthost, repo)
+    rhel_contenthost.execute(f"hostnamectl set-hostname {rhel_contenthost.hostname} --static")
+    rhel_contenthost.install_cockpit()
+    rhel_contenthost.add_rex_key(satellite=default_sat)
+    yield rhel_contenthost
+
+
+def register_host_custom_repo(default_sat, module_org, rhel_contenthost, repo_urls):
+    """Register content host to Satellite and sync repos"""
     # prepare Product and appropriate Satellite Client repo on satellite
     rhelver = rhel_contenthost.os_version.major
     prod = entities.Product(
         organization=module_org, name=f'rhel{rhelver}_{gen_string("alpha")}'
     ).create()
-    sattools_repo = entities.Repository(
-        organization=module_org,
-        product=prod,
-        content_type='yum',
-        url=settings.repos['SATCLIENT_REPO'][f'RHEL{rhelver}'],
-    ).create()
-    sattools_repo.sync()
+    for url in repo_urls:
```

Comment:
why do you need  repo_urls here ?
isn't the one repo per version enough ?

In the previous version you enable one repo per version, 
you don't need more of them right ? for one rhel version one custom satclient repo

Reply:
The cockpit on RHEL 7 requires additional extras, but those need base os RPMs to install correctly.